drivers/src/pci/root.rs

```diff
@@ -55,6 +65,46 @@ impl PciBarType {
             _ => None,
         }
     }
+
+    /// Reads a PCI BAR resource type from the first cell in a PCI address range.
+    fn from_dt_cell(cell: u32) -> Option<Self> {
+        let prefetchable = (cell & PCI_ADDR_PREFETCH_BIT) != 0;
+        use PciBarType::*;
+        match (cell >> PCI_ADDR_SPACE_CODE_SHIFT) & PCI_ADDR_SPACE_CODE_MASK {
+            0x0 => {
+                // Config space. Ignore it since we already got it from 'reg'.
+                None
+            }
+            0x1 => Some(IoPort),
+            0x2 => {
+                if prefetchable {
+                    Some(PrefetchableMem32)
+                } else {
+                    Some(Mem32)
+                }
+            }
+            0x3 => {
+                if prefetchable {
+                    Some(PrefetchableMem64)
+                } else {
+                    Some(Mem64)
+                }
+            }
+            _ => unreachable!(),
+        }
+    }
+
+    /// Returns the PCI address cell used to encode this resource type.
```

Comment:
Nit: Non-public function should probably use regular comment instead of doc-style.

Reply:
Done.